reana_server/rest/secrets.py

```diff
@@ -112,7 +112,10 @@ def add_secrets():  # noqa
               }
     """
     try:
-        user = get_user_from_token(request.args.get("access_token"))
+        if current_user.is_authenticated:
+            user = _get_user_from_invenio_user(current_user.email)
+        else:
+            user = get_user_from_token(request.args.get('access_token'))
```

Comment:
_Issue to be created once merged_.

Investigate usage of OAuth2Server to generate tokens so we do not need to generate and maintain our own, see [OAuth2Server CLI `tokens create`](https://github.com/inveniosoftware/invenio-oauth2server/blob/e3515a9d35f01ddae8fa27d2851ef18c80d96bdc/invenio_oauth2server/cli.py#L63)